### PR TITLE
Add github action for upload gps artifacts to GCS.

### PR DESCRIPTION
**Pull request by guyvelleman**

### CODE REVIEW

**Review by OmriEshhar1 — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/stwo-cairo/155)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @guyvelleman)

___
*[a discussion](https://reviewable.io/reviews/starkware-libs/stwo-cairo/155#-OBKALVQ3mDv7iG1f-be:-OBKALVQ3mDv7iG1f-bf:bdoi222) (no related file):*
Please have someone from Infra (Adi?) to review this as well.

**Review by adi-yakovian-starkware — APPROVED** [review]

File: .github/workflows/upload_artifacts_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    paths:
+      - '.github/workflows/upload_artifacts_workflow.yml'
+
+jobs:
+  artifacts-push:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get commit hash prefix for PR update
```
Comment (adi-yakovian-starkware) [inline .github/workflows/upload_artifacts_workflow.yml:24]: Add the part I told you to remove :grimacing:

**Review by adi-yakovian-starkware — CHANGES_REQUESTED** [review]

**Review by adi-yakovian-starkware — APPROVED** [review]

**Review by adi-yakovian-starkware — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/stwo-cairo/155#-:-OBKe78t1xhIggIcSKaW:bnfp4nl)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @guyvelleman and @OmriEshhar1)

**Review by adi-yakovian-starkware — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/stwo-cairo/155)* status: all files reviewed (commit messages unreviewed), all discussions resolved (waiting on @OmriEshhar1)

**Review by OmriEshhar1 — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/stwo-cairo/155)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @guyvelleman)

**Review by adi-yakovian-starkware — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/stwo-cairo/155)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @guyvelleman)

**Pull request merged**